searchcore/src/vespa/searchcore/proton/attribute/attributemanager.cpp

```diff
@@ -581,6 +581,17 @@ AttributeManager::asyncForEachAttribute(std::shared_ptr<IAttributeFunctor> func)
     }
 }
 
+void
+AttributeManager::asyncForAttribute(const vespalib::string &name, std::shared_ptr<IAttributeFunctor> func) const {
+    AttributeMap::const_iterator itr = _attributes.find(name);
+    if (itr == _attributes.end() || itr->second.isExtra() || !func) {
+        return;
+    }
+    AttributeVector::SP attrsp = itr->second.getAttribute();
+    _attributeFieldWriter.execute(_attributeFieldWriter.getExecutorId(attrsp->getNamePrefix()),
```

Comment:
This is unsafe for two reasons:

attrsp->getNamePrefix() might read freed memory due to attribute flushing occuring at the same time.
_attributeFieldWriter.getExecutorId() might read a hash table that is being modified by the document db master thread.